File: Mar 9 82. Remove Duplicates from Sorted List II (Medium)/Solution.py
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:

        # Create a dummy node
        dummy = ListNode(-1, head)
        prev = dummy
        
        while prev.next and prev.next.next:
            # Init subHead
            subHead = prev.next
            # If not duplicate move prev forward
            if subHead.val != subHead.next.val:
                prev.next = subHead
                prev = prev.next
                continue
            # If duplicate remove them and point prev.next to next integer
            # Note that we not move prev forward at this stage, only remove duplicate
            while subHead.next and subHead.val == subHead.next.val:
                subHead = subHead.next
            prev.next = subHead.next
        # return head
        return dummy.next

File: Mar 9 82. Remove Duplicates from Sorted List II (Medium)/Solution.go
func deleteDuplicates(head *ListNode) *ListNode {
    // Create a dummy node
    dummy := ListNode{0, head}
    prev := &dummy
    
    for prev.Next != nil && prev.Next.Next != nil{
        // Init subHead
        subHead := prev.Next
        // If not duplicate move prev forward
        if subHead.Val != subHead.Next.Val{
            prev.Next = subHead
            prev = prev.Next
            continue
        }
        // If duplicate remove them and point prev.next to next integer
        // Note that we not move prev forward at this stage, only remove duplicate
        for subHead.Next != nil && subHead.Val == subHead.Next.Val{
            subHead = subHead.Next
        }
        prev.Next = subHead.Next
        
    }

    return dummy.Next
}
